modules/core/src/persistence.cpp

```diff
@@ -4025,7 +4025,14 @@ static void
 icvJSONWriteReal( CvFileStorage* fs, const char* key, double value )
 {
     char buf[128];
-    icvJSONWrite( fs, key, icvDoubleToString( buf, value ));
+    int len = (int)strlen( icvDoubleToString( buf, value ) );
```

Comment:
Almost all the uses of strlen in persistence.cpp explicitly cast the return of strlen to an int, even when passed the same pointer as I am (such as line 4823 of original code). I kept this style to remain constant with the rest of the source code.

In both cases, strlen is called for buf which is a static char array of size 128 (first use) or 256 (second use). If a null character in the double-to-string conversion is not found before then, there would have been a more significant error and buffer overflow in the upstream code in any case.

I can make an exception to the previous source style and return the result as a size_t if you prefer - please confirm.